sn_networking/src/error.rs

```diff
@@ -93,3 +92,53 @@ pub enum Error {
     #[error("No SwarmCmd channel capacity")]
     NoSwarmCmdChannelCapacity,
 }
+
+/// Pretty print a `kad::RecordKey` as a hex string.
+/// So clients can use the hex string for xorname and record keys interchangeably.
+/// This makes errors actionable for clients.
+/// The only cost is converting ked::RecordKey into it before sending it in errors: `record_key.into()`
+#[derive(Clone)]
+pub struct PrettyPrintRecordKey(kad::RecordKey);
```

Comment:
this better to be used across all the logs when logging `record_key`, otherwise it won't be tracable along the flow.

ideally, I'd prefer to `overload` the `Debug` trait of the `RecordKey`, however, not sure how to do it in our own crate without raising PR into libp2p crate.

Reply:
Agreed, but we can't overload without wrapping it, will adapt the logs where I find them!